src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -501,8 +504,9 @@ bool isRelevantDocument(SyntaxTreeIndex syntaxTreeInfo)
                 if (deconstructMethods.Any(m => Matches(m, originalUnreducedSymbolDefinition)))
                 {
                     var location = syntaxFacts.GetDeconstructionReferenceLocation(node);
+                    var valueUsageInfo = semanticModel.GetValueUsageInfo(node, semanticFacts, cancellationToken);
```

Comment:
it has document. so might make sense to have exteion method on document then? document.GetValueUsageIfoAsync(node, cancellation)? rahter than one to retrieve things and then call on semanticmodel?

just nit thing since feels like it can save typing.

Reply:
I added it on SemanticModel based on @CyrusNajmabadi suggestion [here](https://github.com/dotnet/roslyn/pull/30123#discussion_r220365191). Eventually it should probably go into the compiler layer.

Reply:
yes.  i would prefer tihs be on semanticmodel.